src/index.js

```diff
@@ -196,6 +196,10 @@ class PubsubBaseProtocol extends EventEmitter {
       protocols: this.multicodecs
     }))
 
+    if (peer.isConnected) {
+      return
+    }
```

Comment:
This looks good, although the logic above is a bit confusing if you don't look at the internals of _addPeer. I was concerned we were creating a new peer unnecessarily but it's just incrementing a reference count if the peer doesn't exist. It's not needed for this PR but `new Peer` should happen inside _addPeer to prevent the unnecessary allocation if we are already tracking the peer.

Reply:
I can change that in a follow up PR as it was older logic